blocks/monitoring/evaluators.py

```diff
@@ -172,7 +195,17 @@ class DatasetEvaluator(object):
 
     """
     def __init__(self, variables, updates=None):
-        self.buffer_ = AggregationBuffer(variables)
+        theano_variables = []
```

Comment:
We should also check that all the names are unique. We have such a check in `AggregationBuffer`, but now it is not sufficient, because there also non-theano monitored quantities.